cpp/src/parquet/bloom_filter.h

```diff
@@ -200,6 +257,11 @@ class PARQUET_EXPORT BlockSplitBloomFilter : public BloomFilter {
 
   bool FindHash(uint64_t hash) const override;
   void InsertHash(uint64_t hash) override;
+  void InsertHashes(const uint64_t* hashes, int num_values) override {
+    for (int i = 0; i < num_values; ++i) {
+      InsertHash(hashes[i]);
```

Comment:
Implement a `InsertHashInternal` to avoid multiple virtual calls here?

Reply:
Emmm I guess I should mark `BlockSplitBloomFilter` as final to avoid virtual call

Reply:
Ok, I will submit a better solution here. But why calling `InsertHash` many times is not ok? I change this filter to "final", so, compiler will try it best to optimizing `InsertHash` without calling virtual function

Reply:
Yes, but now `InsertHashes` is a member of `BlockSplitBloomFilter`, not implemented in `BloomFilter`(base class)

Reply:
Good point. By writing benchmarks you'll be able to evaluate if the two possible implementations show different performance.

Reply:
I change it to calling a non-virtual function, but performance is same